src/runtime/mocker.rs

```diff
@@ -192,10 +237,11 @@ impl<T: Debug + Send + 'static> BufferWriterHost for MockWriter<T> {
         self
     }
 
-    fn produce(&mut self, amount: usize, _tags: Vec<ItemTag>) {
+    fn produce(&mut self, amount: usize, tags: Vec<ItemTag>) {
         unsafe {
             self.data.set_len(self.data.len() + amount);
         }
+        self.tags.extend(tags);
```

Comment:
I didn't test it, but I guess the index of the tags should be offset by `self.data.len()` before it is increased.

Reply:
@bastibl I think this fixes it: [5f1988b](https://github.com/FutureSDR/FutureSDR/pull/168/commits/5f1988beed283aeee1cfdbcd7fce3507436110c9)